salt/modules/win_system.py

```diff
@@ -817,12 +848,15 @@ def set_system_time(newtime):
     :rtype: bool
     '''
     # Get date/time object from newtime
-    dt_obj = salt.utils.date_cast(newtime)
```

Comment:
I can see there are issues with `salt.utils.date_cast` not detecting the correct date/time format. Perhaps we should take a look there. @cachedout